lib/cli/src/commands/compile.rs

```diff
@@ -81,12 +96,49 @@ impl Compile {
         println!("Engine: {}", engine_type.to_string());
         println!("Compiler: {}", compiler_type.to_string());
         println!("Target: {}", target.triple());
+
         let module = Module::from_file(&store, &self.path)?;
         let _ = module.serialize_to_file(&self.output)?;
         eprintln!(
             "✔ File compiled successfully to `{}`.",
             self.output.display(),
         );
+
+        #[cfg(feature = "object-file")]
+        if engine_type == EngineType::ObjectFile {
+            let symbol_registry = module.artifact().symbol_registry();
```

Comment:
Can we cast the artifact to `NativeArtifact` and access the methods there? I don't think all Artifacts should have `symbol_registry` or `metadata_length`